ios/Errors.swift

```diff
@@ -230,26 +230,26 @@ extension ErrorCode.Code {
             return "InvalidAmount"
         case .invalidCurrency:
             return "InvalidCurrency"
-        case .appleBuiltInReaderTOSAcceptanceRequiresiCloudSignIn:
-            return "AppleBuiltInReaderTOSAcceptanceRequiresiCloudSignIn"
-        case .appleBuiltInReaderTOSAcceptanceCanceled:
-            return "AppleBuiltInReaderTOSAcceptanceCanceled"
+        case .tapToPayReaderTOSAcceptanceRequiresiCloudSignIn:
+            return "TapToPayReaderTOSAcceptanceRequiresiCloudSignIn"
+        case .tapToPayReaderTOSAcceptanceCanceled:
+            return "TapToPayReaderTOSAcceptanceCanceled"
         case .nfcDisabled:
             return "NfcDisabled"
         case .readerNotAccessibleInBackground:
             return "ReaderNotAccessibleInBackground"
-        case .appleBuiltInReaderFailedToPrepare:
-            return "AppleBuiltInReaderFailedToPrepare"
-        case .appleBuiltInReaderDeviceBanned:
-            return "AppleBuiltInReaderDeviceBanned"
-        case .appleBuiltInReaderTOSNotYetAccepted:
-            return "AppleBuiltInReaderTOSNotYetAccepted"
-        case .appleBuiltInReaderTOSAcceptanceFailed:
-            return "AppleBuiltInReaderTOSAcceptanceFailed"
-        case .appleBuiltInReaderMerchantBlocked:
-            return "AppleBuiltInReaderMerchantBlocked"
-        case .appleBuiltInReaderInvalidMerchant:
-            return "AppleBuiltInReaderInvalidMerchant"
+        case .tapToPayReaderFailedToPrepare:
+            return "TapToPayReaderFailedToPrepare"
+        case .tapToPayReaderDeviceBanned:
+            return "TapToPayReaderDeviceBanned"
+        case .tapToPayReaderTOSNotYetAccepted:
+            return "TapToPayReaderTOSNotYetAccepted"
+        case .tapToPayReaderTOSAcceptanceFailed:
+            return "TapToPayReaderTOSAcceptanceFailed"
```

Comment:
We've added a new error code to the iOS and Android SDK. (for iOS, it's `SCPErrorGenericReaderError`, and for Android it's `GENERIC_READER_ERROR`). Can we add a new external-facing RN error code for this?

Reply:
In Android we use toString so it doesn't matter if there's new element or not, but on iOS side besides `.genericReaderError` we still have all of these:
```
        case .cancelFailedUnavailable:
            <#code#>
        case .invalidConnectionConfiguration:
            <#code#>
        case .invalidRequiredParameterOnBehalfOf:
            <#code#>
        case .requestDynamicCurrencyConversionRequiresUpdatePaymentIntent:
            <#code#>
        case .dynamicCurrencyConversionNotAvailable:
            <#code#>
        case .surchargeNoticeRequiresUpdatePaymentIntent:
            <#code#>
        case .surchargeUnavailableWithDynamicCurrencyConversion:
            <#code#>
        case .canceledDueToIntegrationError:
            <#code#>
        case .collectInputsInvalidParameter:
            <#code#>
        case .collectInputsUnsupported:
            <#code#>
        case .readerConnectionOfflineNeedsUpdate:
            <#code#>
        case .readerConnectionOfflinePairingUnseenDisabled:
            <#code#>
        case .collectInputsTimedOut:
            <#code#>
        case .usbDiscoveryTimedOut:
            <#code#>
        case .tapToPayReaderAccountDeactivated:
            <#code#>
        case .readerMissingEncryptionKeys:
            <#code#>
        case .usbDisconnected:
            <#code#>
        case .encryptionKeyFailure:
            <#code#>
        case .encryptionKeyStillInitializing:
            <#code#>
        case .collectInputsApplicationError:
            <#code#>
        case .genericReaderError:
            <#code#>
        case .commandInvalidAllowRedisplay:
            <#code#>
        case .onlinePinNotSupportedOffline:
            <#code#>
        case .offlineTestCardInLivemode:
```
Do we need to add them all or only the one you mentioned?

Reply:
I think we want to add all these iOS errors (cc @bric-stripe @mindy-stripe to confirm)

Reply:
Seems we still need some time to merge this, create in separate pr here: https://github.com/stripe/stripe-terminal-react-native/pull/851
Will add new ones here when able to rebase.

Also would like to know if we're consider some ways to automatically convert enum to string without hardcode like Android side?
https://stackoverflow.com/questions/24701075/swift-convert-enum-value-to-string

Reply:
> Also would like to know if we're consider some ways to automatically convert enum to string without hardcode

unfortunately the enum backing `ErrorCode.Code` in the native SDK is an ObjC enum and so can't be declared as being backed by a String.

in the native SDK we do have a code to string helper already and we could expose that, but it wouldn't be available until 4.2 at the earliest.

Reply:
added into this pr now.